infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/batchprocessor/protocolbatchprocessor/ProtocolBatchProcessor.java

```diff
@@ -0,0 +1,187 @@
+package org.corfudb.infrastructure.log.statetransfer.batchprocessor.protocolbatchprocessor;
+
+import com.google.common.collect.Ordering;
+import com.google.common.collect.Sets;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.common.result.Result;
+import org.corfudb.infrastructure.log.StreamLog;
+import org.corfudb.infrastructure.log.statetransfer.batch.Batch;
+import org.corfudb.infrastructure.log.statetransfer.batch.BatchResult;
+import org.corfudb.infrastructure.log.statetransfer.batch.BatchResultData;
+import org.corfudb.infrastructure.log.statetransfer.batchprocessor.BatchProcessorError;
+import org.corfudb.infrastructure.log.statetransfer.batchprocessor.BatchProcessorFailure;
+import org.corfudb.infrastructure.log.statetransfer.batchprocessor.StateTransferBatchProcessor;
+import org.corfudb.protocols.wireprotocol.ILogData;
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.runtime.exceptions.RetryExhaustedException;
+import org.corfudb.runtime.view.AddressSpaceView;
+import org.corfudb.runtime.view.ReadOptions;
+import org.corfudb.util.retry.ExponentialBackoffRetry;
+import org.corfudb.util.retry.IRetry;
+import org.corfudb.util.retry.RetryNeededException;
+
+import java.time.Duration;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * A batch processor that transfers non-committed addresses one batch at a time
+ * via a replication protocol.
+ */
+@Slf4j
+public class ProtocolBatchProcessor implements StateTransferBatchProcessor {
+
+    /**
+     * Configurations for the retry logic.
+     */
+    private static final int MAX_RETRIES = 3;
```

Comment:
please transform these constants to fields, to be able to adjust those parameters